### PR TITLE
AEROGEAR-8601 - Docs for custom metrics implementation

### PR DESCRIPTION
**Pull request by aliok**
https://issues.jboss.org/browse/AEROGEAR-8601

Verification: https://deploy-preview-639--aerogear-docs-preview.netlify.com/aerogear/latest/data-sync/#sync-server-audit-logs

### CODE REVIEW

File: modules/ROOT/pages/_partials/sync/server-audit-logs.adoc [hunk 1]
```diff
@@ -119,6 +120,111 @@ const server = VoyagerServer(apolloConfig, voyagerConfig)
 
 The link:https://github.com/aerogear/voyager-server/blob/master/examples/auditLogging[Audit Logging Example Server Guide] has an example server based off the instructions above and shows all of the steps needed to get it running.
 
+Alternatively, if the default audit logger does not match the requirements, you can implement your own audit logger which implements the `AuditLogger` interface.
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/server-audit-logs.adoc]:
```suggestion
Alternatively, if the default audit logger does not match your requirements, you can create an audit logger that implements the `AuditLogger` interface as defined below.
```

File: modules/ROOT/pages/_partials/sync/server-audit-logs.adoc [hunk 2]
```diff
@@ -119,6 +120,111 @@ const server = VoyagerServer(apolloConfig, voyagerConfig)
...
+
+Definition of the `AuditLogger` interface is below:
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/server-audit-logs.adoc]:
```suggestion
.Definition of the `AuditLogger` interface
```

File: modules/ROOT/pages/_partials/sync/server-audit-logs.adoc [hunk 3]
```diff
@@ -119,6 +120,111 @@ const server = VoyagerServer(apolloConfig, voyagerConfig)
...
+[source,typescript]
+----
+export interface AuditLogger {
+  logResolverCompletion(msg: string, success: boolean, obj: any, args: any, context: any, info: GraphQLResolveInfo): void
+  logConflict (msg: string, serverData: any, clientData: any, obj: any, args: any, context: any, info: GraphQLResolveInfo): void
+  auditLog(msg: string, obj: any, args: any, context: any, info: GraphQLResolveInfo): void
+}
+----
+
+Following example implements an `AuditLogger` and injects it to {sync-server}. This implementation redacts the arguments using a `myCustomRedactionFunction` function.
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/server-audit-logs.adoc]:
```suggestion
The following example implements an `AuditLogger` and injects it into the {sync-server}. 
The example redacts the arguments using a `myCustomRedactionFunction` function.
```

File: modules/ROOT/pages/_partials/sync/server-audit-logs.adoc [hunk 4]
```diff
@@ -119,6 +120,111 @@ const server = VoyagerServer(apolloConfig, voyagerConfig)
...
+const voyagerConfig = {
+  auditLogger
+}
+const server = VoyagerServer(apolloConfig, voyagerConfig)
+
+----
+
 = Sending Device Information in {sync-client}
 
 Please see <<#sync-clien-audit-logs, {sync-client} Audit Logs>> section for more details.
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/server-audit-logs.adoc]:
```suggestion
See the <<#sync-clien-audit-logs, {sync-client} Audit Logs>> section for more information.
```

File: modules/ROOT/pages/_partials/sync/server-audit-logs.adoc [hunk 5]
```diff
@@ -107,7 +108,7 @@ Audit logging is enabled in {sync-server} using the link:https://www.npmjs.com/p
 const auditLogger = require('@aerogear/voyager-audit')
 ----
 
-3. Inject the auditLogger module into {sync-server}. This will enable audit logging within your application.
+2. Inject the auditLogger module into {sync-server}. This will enable audit logging within your application.
```
Comment (finp) [inline modules/ROOT/pages/_partials/sync/server-audit-logs.adoc]:
```suggestion
2. Inject the auditLogger module into the {sync-server}. This enables audit logging within your application.
```

**Comment by aliok** [comment]
@finp Thanks for the awesome suggestions. It worked really nicely. Just pressed "Commit changes" button a few times :)

**Comment by aliok** [comment]
No, thanks. Merging now

**Pull request merged**